src/TAPVerifier.sol

```diff
@@ -33,34 +33,33 @@ contract TAPVerifier is EIP712 {
 
     /**
      * @dev Recovers the signer address of a signed ReceiptAggregationVoucher (RAV).
-     * @param _signedRAV The SignedRAV containing the RAV and its signature.
+     * @param signedRAV The SignedRAV containing the RAV and its signature.
      * @return The address of the signer.
      * @notice REVERT: This function may revert if ECDSA.recover fails, check ECDSA library for details.
      */
-    function recoverRAVSigner(SignedRAV memory _signedRAV) public view returns (address) {
-        bytes32 messageHash = hashRAV(_signedRAV.rav);
-        return ECDSA.recover(messageHash, _signedRAV.signature);
+    function recoverRAVSigner(SignedRAV memory signedRAV) public view returns (address) {
```

Comment:
All `memory` instances in this contract could be `calldata` and save a bit of gas.